docker-compose.yml

```diff
@@ -1,25 +1,39 @@
-version: '3.5'
+version: "3.7"
 
 services:
-
   api:
     container_name: nest_api
-    build: .
+    build:
+      context: .
+      target: development
+    volumes:
+      - .:/usr/src/app
+      - /usr/src/app/node_modules
     ports:
       - ${API_PORT}:${API_PORT}
-    volumes:
-      - .:/home/api
-      - /home/api/node_modules
```

Comment:
Aqui esta sendo feito um override das variáveis que o compose pega do arquivo .env pra forçar que quando a api rodar com o docker-compose, ela sempre pegue a porta e o host do database que esta no compose também. Dessa forma se mudarmos a porta do banco no .env ou o host pra rodarmos localmente, não irá dar problema

Reply:
Oi Yuri! Não entendi seu comentário. Primeiro que esse é um pull request com algumas alterações sugeridas e no tocante ao mapeamento de volumes e portas está exatamente igual. A mesma ideia. Veja:

Original:
 ```
    ports:
      - ${API_PORT}:${API_PORT}
    volumes:
      - .:/home/api
      - /home/api/node_modules
```

Modificada:
```
     volumes:
      - .:/usr/src/app
      - /usr/src/app/node_modules
    ports:
      - ${API_PORT}:${API_PORT}
      - ${API_PORT}:${API_PORT}
```

Em relação à dá conflito ao rodar no ambiente local, é só rodar ```docker-compose down``` e a ideia de trabalhar com docker é exatamente desenvolver usando ele, por isso, essa não deveria ser uma preocupação inicialmente.

Reply:
Opa! eu estava falando sobre as linhas 13, 14 e 15 onde ta sendo setado o DB_HOST e o DB_PORT. Só quis deixar claro que nesse caso eu coloquei elas pra funcionar quando a api for rodar local sem o docker sem ter ficar alternando o host entre por exemplo "localhost" e "database". O mesmo para a porta. Porque nesses 2 casos como tu pode conferir no docker-compose estou usando o .env apenas nas portas que estao sendo mapeadas para o host.